src/init.cpp

```diff
@@ -1459,8 +1459,10 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
                 strLoadError = _("Error initializing block database");
                 break;
             case ChainstateLoadingError::ERROR_CHAINSTATE_UPGRADE_FAILED:
-                strLoadError = _("Error upgrading chainstate database");
-                break;
+                return InitError(_("Unsupported chainstate database format found. "
+                                   "Please restart with -reindex. This will redownload "
```

Comment:
Isn't  `-reindex-chainstate` enough?
Also, `-reindex` will not redownload the blockchain. It only reindexes the one on disk.

Reply:
No idea. For me Bitcoin Core just hangs itself up when using -reindex-chainstate and -prune, but that's probably an unrelated bug.

Reply:
Reported as https://github.com/bitcoin/bitcoin/issues/24242

Reply:
Picked your suggestion, thx.

Reply:
It wasn't clear to me that the context here was about pruning. But that definitely sounds like a bug. With pruning, `-reindex-chainstate` and `-reindex` should be the same.